src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -176,25 +191,76 @@ public Action createAction(final Action action) {
   }
 
   @Transactional
-  public void createScheduledActions(Integer actionPlanJobId) {
-    final ActionPlanJob actionPlanJob =
-        actionPlanJobRepository.findByActionPlanJobPK(actionPlanJobId);
+  public void createScheduledActions(ActionPlan actionPlan, ActionPlanJob actionPlanJob) {
+    List<ActionCase> cases = actionCaseRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
+    List<ActionRule> rules = actionRuleRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
 
-    if (actionPlanJob == null) {
-      return;
+    // For each case/rule pair create an action
+    cases.forEach(caze -> createActionsForCase(caze, rules));
+    updatePlanAndJob(actionPlan, actionPlanJob);
+  }
+
+  private void createActionsForCase(ActionCase actionCase, List<ActionRule> actionRules) {
+    if (isActionPlanLive(actionCase)) {
+      actionRules.forEach(rule -> createActionForCaseAndRule(actionCase, rule));
+    }
+  }
+
+  private boolean isActionPlanLive(ActionCase actionCase) {
+    final Timestamp currentTime = new Timestamp((new Date()).getTime());
+    return actionCase.getActionPlanStartDate().before(currentTime)
+        && actionCase.getActionPlanEndDate().after(currentTime);
+  }
+
+  private void createActionForCaseAndRule(ActionCase actionCase, ActionRule actionRule) {
+    if (hasRuleTriggered(actionRule)) {
+      try {
+        createAction(actionCase, actionRule);
+      } catch (Exception ex) {
```

Comment:
I'm always pretty afraid of any catch-all exception handler. Any reason why we have this here?